resources/qml/Avatar.qml

```diff
@@ -35,10 +36,29 @@ Rectangle {
         font.pixelSize: avatar.height / 2
         verticalAlignment: Text.AlignVCenter
         horizontalAlignment: Text.AlignHCenter
-        visible: img.status != Image.Ready
+        visible: img.status != Image.Ready && !Settings.useIdenticon
         color: Nheko.colors.text
     }
 
+    Image {
+        id: identicon
+        anchors.fill: parent
+        visible: img.status != Image.Ready && Settings.useIdenticon
```

Comment:
Can you swap the order around? That way it doesn't check the image status at all, if identicons are disabled. The setting should change less often.